### PR TITLE
Fix doc links in `std::cmp` module

### PR DESCRIPTION
**Pull request by tesuji**
r? @jonas-schievink

### CODE REVIEW

**Comment by lzutao** [comment]
For some reasons, the doc links which are fine in `core::cmp` but in `std::cmp` they are not rendered:
image

**Comment by jonas-schievink** [comment]
> For some reasons, the doc links which are fine in `core::cmp` but in `std::cmp` they are not rendered:

That's probably due to https://github.com/rust-lang/rust/issues/59833. Are both `std` and `core` links now working after this?

**Comment by lzutao** [comment]
Yes, they are both working after this PR. I would prefer intra-links issue fixed, but I don't 
have strong opinion about it.
I would let the decision to you or other reviewers.

**Comment by jonas-schievink** [comment]
Okay, nice. Fixed links are always better than broken ones, so

@bors r+ rollup

**Comment by lzutao** [comment]
Hm, Wait! I've just seen that there are 2 Eq identifiers in `std::cmp`:
image

Which one should I choose to document?

**Comment by jonas-schievink** [comment]
The traits should be the correct ones to document here

**Comment by lzutao** [comment]
I changed them to point to trait items.

File: src/libcore/cmp.rs [hunk 1]
```diff
@@ -9,14 +9,22 @@
...
 //! For more details, see the respective documentation of each item in the list.
+//!
+//! [`Eq`]: trait.Eq.html
+//! [`PartialEq`]: trait.PartialEq.html
+//! [`Ord`]: trait.Ord.html
+//! [`PartialOrd`]: trait.PartialOrd.html
+//! [`Ordering`]: enum.Ordering.html
+//! [`Reverse`]: struct.Reverse.html
+//! [`max`]: fn.max.html
+//! [`min`]: fn.max.html
```
Comment (ehuss) [inline src/libcore/cmp.rs]:
```suggestion
//! [`min`]: fn.min.html
```

?

Comment (lzutao) [reply src/libcore/cmp.rs]: Oops. Fixed.

**Pull request merged**